src/Guesser/TypeGuesser.php

```diff
@@ -76,36 +77,36 @@ public function guessType($class, $property, ModelManagerInterface $modelManager
             case 'simple_array':
             case 'json':
             case 'json_array':
-                return new TypeGuess(TemplateRegistry::TYPE_ARRAY, [], Guess::HIGH_CONFIDENCE);
+                return new TypeGuess(FieldDescriptionInterface::TYPE_ARRAY, [], Guess::HIGH_CONFIDENCE);
             case 'boolean':
-                return new TypeGuess(TemplateRegistry::TYPE_BOOLEAN, [], Guess::HIGH_CONFIDENCE);
+                return new TypeGuess(FieldDescriptionInterface::TYPE_BOOLEAN, [], Guess::HIGH_CONFIDENCE);
             case 'datetime':
             case 'datetime_immutable':
             case 'vardatetime':
             case 'datetimetz':
             case 'datetimetz_immutable':
-                return new TypeGuess(TemplateRegistry::TYPE_DATETIME, [], Guess::HIGH_CONFIDENCE);
+                return new TypeGuess(FieldDescriptionInterface::TYPE_DATETIME, [], Guess::HIGH_CONFIDENCE);
             case 'date':
             case 'date_immutable':
-                return new TypeGuess(TemplateRegistry::TYPE_DATE, [], Guess::HIGH_CONFIDENCE);
+                return new TypeGuess(FieldDescriptionInterface::TYPE_DATE, [], Guess::HIGH_CONFIDENCE);
             case 'decimal':
             case 'float':
-                // NEXT_MAJOR: return new TypeGuess(TemplateRegistry::TYPE_FLOAT, [], Guess::LOW_CONFIDENCE)
+                // NEXT_MAJOR: return new TypeGuess(FieldDescriptionInterface::TYPE_FLOAT, [], Guess::LOW_CONFIDENCE)
                 return new TypeGuess('number', [], Guess::MEDIUM_CONFIDENCE);
             case 'integer':
             case 'bigint':
             case 'smallint':
-                return new TypeGuess(TemplateRegistry::TYPE_INTEGER, [], Guess::MEDIUM_CONFIDENCE);
+                return new TypeGuess(FieldDescriptionInterface::TYPE_INTEGER, [], Guess::MEDIUM_CONFIDENCE);
             case 'string':
-                return new TypeGuess(TemplateRegistry::TYPE_TEXT, [], Guess::MEDIUM_CONFIDENCE);
+                return new TypeGuess(FieldDescriptionInterface::TYPE_TEXT, [], Guess::MEDIUM_CONFIDENCE);
```

Comment:
this one actually does not exist on 3.x

Reply:
Should use `TYPE_STRING` I think?

https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Templating/TemplateRegistryInterface.php#L67

Reply:
TYPE_STRING is `'string'` when this one is `'text'`.
We have to keep the deprecated one on 3.x and change it on master then.